libvast/vast/concept/printable/std/chrono.hpp

```diff
@@ -82,31 +105,61 @@ struct duration_printer : printer<duration_printer<Rep, Period, Policy>> {
   }
 };
 
+namespace {
+
+struct year_month_day {
+  unsigned short year;
+  unsigned char month;
+  unsigned char day;
+};
+
+inline year_month_day from_days(days dp) noexcept {
+  static_assert(std::numeric_limits<unsigned>::digits >= 18,
+                "This algorithm has not been ported to a 16 bit unsigned "
+                "integer");
+  static_assert(std::numeric_limits<int>::digits >= 20,
+                "This algorithm has not been ported to a 16 bit signed "
+                "integer");
+  auto const z = dp.count() + 719468;
+  auto const era = (z >= 0 ? z : z - 146096) / 146097;
+  auto const doe = static_cast<unsigned>(z - era * 146097); // [0, 146096]
+  auto const yoe = (doe - doe / 1460 + doe / 36524 - doe / 146096)
+                   / 365; // [0, 399]
+  auto const y = static_cast<days::rep>(yoe) + era * 400;
+  auto const doy = doe - (365 * yoe + yoe / 4 - yoe / 100); // [0, 365]
+  auto const mp = (5 * doy + 2) / 153;                      // [0, 11]
+  auto const d = static_cast<unsigned char>(doy - (153 * mp + 2) / 5
+                                            + 1); // [1, 31]
+  auto const m = static_cast<unsigned char>(mp < 10 ? mp + 3
+                                                    : mp - 9); // [1, 12]
+  return year_month_day{static_cast<unsigned short>(y + (m <= 2)), m, d};
+}
```

Comment:
Also here it would be good to have an external link in the documentation for folks that read this code for the first time.